app/src/main/java/com/nervousfish/nervousfish/modules/database/KeyAdapter.java

```diff
@@ -0,0 +1,85 @@
+package com.nervousfish.nervousfish.modules.database;
+
+import com.google.gson.TypeAdapter;
+import com.google.gson.stream.JsonReader;
+import com.google.gson.stream.JsonWriter;
+import com.nervousfish.nervousfish.data_objects.IKey;
+import com.nervousfish.nervousfish.data_objects.RSAKey;
+import com.nervousfish.nervousfish.data_objects.SimpleKey;
+
+import java.io.IOException;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * Adaptor for the {@link IKey} interface for the GSON library.
+ */
+final class KeyAdapter extends TypeAdapter<IKey> {
+
+    /**
+     * Constructor for the {@code KeyAdapter} class.
+     */
+    KeyAdapter() {
+        super();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void write(final JsonWriter writer, final IKey key) throws IOException {
```

Comment:
I don't think so since this class is solely intended to read/write to JSON files. Maybe we can change the classname to indicate this, e.g. `GsonKeyAdapter`. What do you think?